sips/sip-0xx/sip-0xx-nakamoto.md

```diff
@@ -0,0 +1,658 @@
+   - If the number of remaining addresses in the reward set, N, is less than 20,
+     then the producer set enrollment must burn BTC by including (20-N) burn
+     outputs
+2. Otherwise, the second output must be a burn address.
+
+During a reward cycle, this enrollment transaction will include a somewhat large
+number of outputs: one `OP_RETURN`, twenty stacker rewards, and one change
+address, totaling 22 outputs. While this might seem like a substantial
+transaction, it effectively replaces ten separate transactions under the SIP-007
+leader block commit scheme, each of which would have four outputs (one
+`OP_RETURN`, two stacker rewards, and one change address). Furthermore, the
+enrollment window's duration of ten blocks potentially allows would-be producers
+to take advantage of lower transaction fees during one of those blocks. Despite
+the higher fee for this larger transaction, the cost can be spread out or
+amortized across the ten blocks of the set, resulting in a lower overall cost
+compared to the previous system.
+
+### Censorship Resistance
+
+The producer set enrollments for set _N_ can be included in any of the 10
+Bitcoin blocks in producer set _N-2_. This makes it extremely difficult for a
+Bitcoin miner to censor these transactions, since to do so, they would need to
+control all 10 Bitcoin blocks in that term.
+
+### Selecting Producers
+
+Would-be producers with valid producer set enrollments in term _N-2_ are
+eligible to be included in the producer set for term _N_. The total number of
+producers in a set needs to be limited to prevent coinbase payouts from
+including too many accounts, as this could slow down event processing and even
+open a new DoS vector. This cap will also prevent the block-signing process from
+becoming too expensive. To this end, the total amount of BTC spent in the
+outputs described in "Producer Set Enrollments" is used to select the producer
+set. Would-be producers are ranked by these BTC expenditures, and the top 100
+will be selected for the producer set.
+
+## Producer Rewards
+
+During a term, producers in the set are eligible to receive a portion of the
+coinbase rewards and transaction fees for blocks they produce. Since a term is
+defined as 10 Bitcoin blocks, the coinbase reward is equal to 10 times the
+coinbase as defined in **_<which SIP defines this?>_**. This amount is
+distributed to all producers in the set proportionally, based on the percentage
+of the total BTC spent in the producer set enrollments. All producers receive
+their portion of the coinbase, regardless of whether or not they signed the
+blocks produced by the set. The coinbase transaction should be the first
+transaction in a term.
+
+The producer set is then incentivized to continue producing blocks throughout
+the term by the transaction fees. Transaction fees are paid only to producer set
+participants who signed the blocks produced. For each block, _B_, the total BTC
```

Comment:
Does this incentivize some kind of race to submit a block to the stackers to sign before all of the producers have had a chance to sign it (after 67% have signed)?

Reply:
I think it does incentivize trying to minimize the set of signers, yes. I'm not sure if this is a time-race, though.